aws-lc-rs/src/aead.rs

```diff
@@ -433,6 +441,149 @@ impl<N: NonceSequence> SealingKey<N> {
             .seal_in_place_separate_tag(Some(self.nonce_sequence.advance()?), aad.as_ref(), in_out)
             .map(|(_, tag)| tag)
     }
+
+    /// Returns a `SealingKeyOpMut` with the next computed `Nonce` from the `NonceSequence` for the next operation.
+    /// # Errors
+    /// `Unspecified` if there is a failure computing the nonce for the next operation.
+    pub fn prepare_operation(&mut self) -> Result<SealingKeyOpMut<'_, N>, Unspecified> {
+        SealingKeyOpMut::new(self)
+    }
+}
+
+macro_rules! nonce_seq_key_op_mut {
+    ($name:ident) => {
+        paste! {
+        /// A key operation with a precomputed nonce from a key's associated `NonceSequence`.
+        pub struct [<$name OpMut>]<'a, N: NonceSequence> {
+            key: &'a mut $name<N>,
+            nonce: Nonce,
+        }
+
+        impl<'a, N: NonceSequence> [<$name OpMut>]<'a, N> {
+            fn new(key: &'a mut $name<N>) -> Result<Self, Unspecified> {
+                let nonce = key.nonce_sequence.advance()?;
+                Ok(Self {
+                    key,
+                    nonce,
+                })
+            }
+        }
+
+        impl<'a, N: NonceSequence> Debug for [<$name OpMut>]<'a, N> {
+            fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> Result<(), core::fmt::Error> {
+                f.debug_struct(stringify!([<$name OpMut>])).finish_non_exhaustive()
+            }
+        }
+        }
+    };
+}
+
+nonce_seq_key_op_mut!(OpeningKey);
+nonce_seq_key_op_mut!(SealingKey);
+
+impl<N: NonceSequence> OpeningKeyOpMut<'_, N> {
+    /// Returns the Nonce that will be used for this operation.
+    #[must_use]
+    pub fn nonce(&self) -> Nonce {
```

Comment:
It would be more performant to just return a reference to the internal `Nonce`:
```
    #[must_use]
    pub fn nonce(&self) -> &Nonce {
        &self.nonce
    }
```